File: meetcode/arrays/Longest Consecutive Sequence.py
# https://leetcode.com/problems/longest-consecutive-sequence/description/

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0

        if len(nums) == 1:
            return 1
        
        nums.sort()
        maxRef = 1
        curr = 1

        # length of nums is over two 
        for i in range(1,len(nums)):
            if nums[i] == nums[i-1]:
                continue
            elif nums[i] == nums[i-1] + 1:
                curr += 1
            else:
                curr = 1
            maxRef = max(maxRef, curr)

        return maxRef

File: neetcode/go/Arrays & Hashing/Longest Consecutive Sequence.go
// https://neetcode.io/problems/longest-consecutive-sequence?list=neetcode150

func longestConsecutive(nums []int) int {
    numSet := make(map[int]struct{})
    for _, num := range nums {
        numSet[num] = struct{}{}
    }

    longest := 0
    for num := range numSet {
        if _, found := numSet[num-1]; !found {
            length := 1
            for {
                if _, exists := numSet[num+length]; exists {
                    length++
                } else {
                    break
                }
            }
            if length > longest {
                longest = length
            }
        }
    }
    return longest
}